src/res/enkf/ert_run_context.py

```diff
@@ -1,103 +1,44 @@
-    _get_iter = ResPrototype("int ert_run_context_get_iter( ert_run_context )")
-    _get_target_fs = ResPrototype(
-        "enkf_fs_ref ert_run_context_get_update_target_fs( ert_run_context )"
-    )
-    _get_sim_fs = ResPrototype(
-        "enkf_fs_ref ert_run_context_get_sim_fs( ert_run_context )"
-    )
-    _get_init_mode = ResPrototype(
-        "enkf_init_mode_enum ert_run_context_get_init_mode( ert_run_context )"
-    )
-
-    _get_step = ResPrototype("int ert_run_context_get_step1(ert_run_context)")
-    _deactivate_realization = ResPrototype(
-        "void ert_run_context_deactivate_realization( ert_run_context, int)"
-    )
-
-    def __init__(
-        self,
-        run_type: EnkfRunType,
-        sim_fs: Optional[EnkfFs],
-        target_fs: Optional[EnkfFs],
-        mask: List[bool],
-        paths: List[str],
-        jobnames: List[str],
-        itr: int = 0,
-        init_mode=EnkfInitModeEnum.INIT_CONDITIONAL,
-    ):
-        c_ptr = self._alloc(
-            run_type,
-            init_mode,
-            itr,
-        )
-        super().__init__(c_ptr)
-
-        _lib.ert_run_context.set_active(self, mask)
-        if sim_fs is not None:
-            _lib.ert_run_context.set_sim_fs(self, sim_fs)
-        if target_fs is not None:
-            _lib.ert_run_context.set_target_fs(self, target_fs)
-
-        if run_type == EnkfRunType.ENSEMBLE_EXPERIMENT:
-            _lib.ert_run_context.add_ensemble_experiment_args(self, paths, jobnames)
-        elif run_type == EnkfRunType.SMOOTHER_RUN:
-            _lib.ert_run_context.add_smoother_run_args(
-                self,
-                paths,
-                jobnames,
-            )
-        elif run_type == EnkfRunType.INIT_ONLY:
-            _lib.ert_run_context.add_init_only_args(self, paths)
-        elif run_type == EnkfRunType.SMOOTHER_UPDATE:
-            pass
-        elif run_type == EnkfRunType.CASE_INIT_ONLY:
-            pass
-        else:
-            raise ValueError(f"Unsupported run type {run_type}")
+@dataclass
+class ErtRunContext:
+    sim_fs: Optional[EnkfFs]
+    target_fs: Optional[EnkfFs]
+    mask: List[bool]
+    paths: List[str]
+    jobnames: Optional[List[str]]
+    itr: int = 0
+    init_mode: EnkfInitModeEnum.INIT_CONDITIONAL = EnkfInitModeEnum.INIT_CONDITIONAL
+
+    def __post_init__(self):
+        self.run_id = f"{uuid.uuid4()}:{datetime.now().strftime('%Y-%m-%dT%H%M')}"
+        self.run_args = []
+        if self.jobnames and self.paths:
+            for iens, (job_name, path) in enumerate(zip(self.jobnames, self.paths)):
+                self.run_args.append(
+                    RunArg(
+                        str(self.run_id),
+                        self.sim_fs,
+                        iens,
+                        self.itr,
+                        path,
+                        job_name,
+                    )
+                )
 
     @classmethod
     def ensemble_experiment(
```

Comment:
Could you start on a new PR to replace uses of this classmethod with the constructor while you are in this area of the code?

Reply:
Yes, that makes a lot of sense! Will create an issue and pick it up to make it visible which changes are being made.